aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -160,165 +177,147 @@ module post_mint_reveal_nft::minting {
     /// The source certificate id not found in the signer's account.
     const ETOKEN_ID_NOT_FOUND: u64 = 15;
     /// Can only exchange after the reveal starts.
-    const ECANNOT_EXCHANGE_BEFORE_REVEAL_SRARTS: u64 = 16;
+    const ECANNOT_EXCHANGE_BEFORE_REVEAL_STARTS: u64 = 16;
     /// Can only add unique token uris.
     const EDUPLICATE_TOKEN_URI: u64 = 17;
+    /// The whitelist stage provided is not valid.
+    const EINVALID_STAGE: u64 = 18;
 
     /// Initialize NFTMintConfig for this module.
-    fun init_module(post_mint_reveal_nft_resource_account: &signer) {
-        let resource_signer_cap = resource_account::retrieve_resource_account_cap(post_mint_reveal_nft_resource_account, @source_addr);
-        move_to(post_mint_reveal_nft_resource_account, NFTMintConfig {
+    fun init_module(admin: &signer) {
+        let seed_vec = bcs::to_bytes(&signer::address_of(admin));
+        vector::append(&mut seed_vec, bcs::to_bytes(&timestamp::now_seconds()));
+        let (_, resource_signer_cap) = account::create_resource_account(admin, seed_vec);
+
+        move_to(admin, NFTMintConfig {
             // The initial admin account will be the source account (which created the resource account);
             // The source account can then update the admin account in NFTMintConfig struct by calling set_admin().
-            admin: @source_addr,
-            treasury: @source_addr,
+            treasury: @post_mint_reveal_nft,
             signer_cap: resource_signer_cap,
-            token_minting_events: account::new_event_handle<MintingEvent>(post_mint_reveal_nft_resource_account),
-            token_exchange_events: account::new_event_handle<ExchangeEvent>(post_mint_reveal_nft_resource_account),
+            token_minting_events: account::new_event_handle<MintingEvent>(admin),
+            token_exchange_events: account::new_event_handle<ExchangeEvent>(admin),
         });
     }
 
-    /// Set admin of this module.
-    public entry fun set_admin(admin: &signer, new_admin_address: address) acquires NFTMintConfig {
-        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
-        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
-        nft_mint_config.admin = new_admin_address;
-    }
-
     /// Set the treasury account (where the payment for NFT goes to) of this module.
     public entry fun set_treasury(admin: &signer, new_treasury_address: address) acquires NFTMintConfig {
+        assert!(signer::address_of(admin) == @post_mint_reveal_nft, error::permission_denied(ENOT_AUTHORIZED));
         let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
-        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
         nft_mint_config.treasury = new_treasury_address;
```

Comment:
Can you verify that the treasury exists and is registered to receive APT? You can use aptos_account::assert_account_is_registered_for_apt